Stage II/Single_Year_Stage_II.py

```diff
@@ -13,6 +13,10 @@ def Single_Year_Stage_II(puf, Stage_I_factors, Stage_II_targets, year, tol):
 
     print("Preparing coefficient matrix...")
 
+    # the first half of this function (with all of the np.where statements)
+    # might be amenable to refactoring to a numba function, could improve
+    # readability
```

Comment:
Oh I did not think of `@vectorize`, although that may be a good idea. I meant that it may be possible to use @jit or perhaps even our custom @iterate_jit (or a similar technique) to expand the np.wheres to to be 'if' statements and so on.